tests/kani/Intrinsics/ConstEval/type_id.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Check that the `type_id` intrinsic is supported with common data types
+#![feature(core_intrinsics)]
+use std::intrinsics::type_id;
+
+struct MyStruct {}
+
+enum MyEnum {}
+
+#[kani::proof]
+fn main() {
+    // Scalar types
+    let _ = type_id::<i8>();
+    let _ = type_id::<i16>();
```

Comment:
Would adding some asserts make sense, e.g.
```rust
let i8_tid = type_id::<i8>();
let i16_tid = type_id::<i16>();
assert_ne!(i8_tid, i16_tid)
```
?
If so, perhaps add one assert for every two consecutive types (one for (i8, i16), one for (i16, i32), etc.).